src/org/jetbrains/sbt/project/ExternalSourceRootResolution.scala

```diff
@@ -83,24 +94,33 @@ trait ExternalSourceRootResolution { self: SbtProjectResolver =>
     moduleNode
   }
 
-  private def externalSourceRootGroupsIn(projects: Seq[Project]): Seq[RootGroup] = {
+  private def scopeAndKindToSourceType(scope: Root.Scope, kind: Root.Kind): ExternalSystemSourceType =
+    (scope, kind) match {
+      case (Root.Scope.Compile, Root.Kind.Sources)    => ExternalSystemSourceType.SOURCE
+      case (Root.Scope.Compile, Root.Kind.Resources)  => ExternalSystemSourceType.RESOURCE
+      case (Root.Scope.Test, Root.Kind.Sources)       => ExternalSystemSourceType.TEST
+      case (Root.Scope.Test, Root.Kind.Resources)     => ExternalSystemSourceType.TEST_RESOURCE
+    }
+
+  private def sharedAndExternalRootsIn(projects: Seq[Project]): Seq[SharedRoot] = {
     val projectRoots = projects.flatMap(project => sourceRootsIn(project).map(ProjectRoot(project,_)))
 
     // TODO return the message about omitted directories
     val internalSourceDirectories = projectRoots.filter(_.isInternal).map(_.root.directory).toSeq
 
-    val sharedRoots = projectRoots
-            .filter(it => it.isExternal && !internalSourceDirectories.contains(it.root.directory))
-            .groupBy(_.root)
-            .mapValues(_.map(_.project).toSet)
-            .map(p => SharedRoot(p._1, p._2.toSeq))
-            .toSeq
-
+    projectRoots
+      .filter(it => it.isExternal && !internalSourceDirectories.contains(it.root.directory))
```

Comment:
!internalSourceDirectories.contains will exclude all roots that are "shared" with other projects. That's good for the project model, but not for the user warning.